lib/spack/spack/build_systems/windows_variants.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.directives import conflicts, variant
+from spack.package import PackageBase
+
+
+class WindowsPackage(PackageBase):
+    """Auxiliary class for managing Windows-specific variants of
+    packages. This class establishes three major variants for use in
+    individual package.py scripts:
+
+        1. Static with links to the dynamic runtime library (/MD)
+        2. Shared builds which link to /MD by necessity
+        3. Static with links to the static runtime library (/MT)
+
+    Variant 1, static with dynamic runtime, will be the default build
+    and the other two (shared and staticmt) can be enabled by the
+    user from the command line with behaviors varying depending on the
+    package.
+    """
+
+    variant('staticmt', default=False,
+            description="Build static version with static runtime libraries")
+
+    conflicts('+staticmt', when='platform=darwin',
```

Comment:
You can remove the `conflicts` here if you do:

```
with when(platform=windows):
  variant(staticmt)
```

In which case the `staticmt` variant won't exist (+ or -) on systems other than Windows.

I'm still a bit concerned about the possibility of mixing this variant with `static/shared` variants on existing package (i.e. if the package already defines a `shared` and inherits from this) but I haven't yet examined the packages in this PR which inherit `WindowsPackage` to see if that will be an issue.

Reply:
Done